### ISSUE TITLE
Problem with d3.behavior.zoom (Mac OS)

### ISSUE THREAD

**Issue by roveo** [mbostock/d3#2205]
The page loads, then you pan to either side and then zoom, and it throws a javascript error in console. This is strange because just today I was using zoom behavior in Windows, Chrome and (as far as I remember) it worked.

__The problem only exists if you use 3-finger trackpad gesture for dragging and don't change cursor position after panning.__ Doesn't work if you drag with hard-click. So, pan with 3 fingers and try to zoom right away. Seems like d3 can't determine cursor position after dragging.

Errors
Safari: `TypeError: null is not an object (evaluating 'p[0]')` line 1228
Chrome:
```
Uncaught TypeError: Cannot read property '0' of null    d3.js:1382

translateTo    d3.js:1382
mousewheeled    d3.js:1519
(anonymous function)    d3.js:1084
```
Tested in both browsers on Mac OS.

Addition:
Recreated the same in Firefox. Firefox and Chrome are pointing to this bit of code (lines 1380-1384):
```js
function translateTo(p, l) {
      l = point(l);
      view.x += p[0] - l[0];
      view.y += p[1] - l[1];
    }
```
Here's a short example:

```html
<!doctype html>
<html>
    <head>
        <script type="text/javascript" src="d3.js"></script>
        <script type="text/javascript">
            window.onload = function () {
                var svg = d3.select("body")
                    .append("svg")
                    .attr({
                        width: 640,
                        height: 480
                    });
                var data = d3.range(10);
                var x = d3.scale.linear()
                    .domain(d3.extent(data))
                    .range([30, 610]);
                var zoom = d3.behavior.zoom()
                    .x(x)
                    .scaleExtent([1, 10])
                    .on("zoom", function () {
                        d3.selectAll("circle").attr("cx", function (d) { return x(d); });
                    });
                var g = svg.append("g").call(zoom);
                g.append("rect")
                    .attr({
                        x: 0, y: 0,
                        width: 640,
                        height: 480,
                        opacity: 0
                    })
                g.selectAll("circle")
                    .data(data).enter()
                    .append("circle")
                    .attr({
                        cx: function (d) { return x(d); },
                        cy: 50,
                        r: 10,
                        stroke: "none",
                        fill: "black"
                    });
            }
        </script>
    </head>
    <body></body>
</html>
```

**Comment by roveo** [comment]
Recreated the same in Firefox. Firefox and Chrome are pointing to this bit of code (lines 1380-1384):
```js
function translateTo(p, l) {
      l = point(l);
      view.x += p[0] - l[0];
      view.y += p[1] - l[1];
    }
```

**Comment by nicrizzo** [comment]
I found the same problem here http://bl.ocks.org/mbostock/3892928
It can be reproduced with a mouse wheel (tested on mac os x 10.9.5 and yosemite): just press and release the wheel when scrolling.
I changed the function mousewheeled:
<pre>center0 && translateTo(center0, translate0);</pre>
(instead of <pre>translateTo(center0, translate0)</pre>)
Quick and dirty, but it works

**Comment by ilyabo** [comment]
Same problem can be experienced with this vector tiles example http://bl.ocks.org/mbostock/5593150 after doing normal panning and zooming for a while. 

I noticed that in function `mousewheeled()` the variable `center0` is initialized only when  `mousewheelTimer` is falsy:

     if (mousewheelTimer) clearTimeout(mousewheelTimer); else translate0 = location(center0 = center || d3.mouse(this))

The error comes when `mousewheelTimer` is truthy,  `center0` is not initialized, but is used in ` translateTo(center0, translate0)`.

**Comment by jasondavies** [comment]
Proposed fix in #2351.
